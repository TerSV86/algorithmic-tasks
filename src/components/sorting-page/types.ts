export interface Step {
    currentArray: number[];
    sortedIndexes: number[];
    aIndex?: number;
    bIndex?: number;
}

export enum SortKind {
    Select = "select",
    Bubble = 'bubble'
}