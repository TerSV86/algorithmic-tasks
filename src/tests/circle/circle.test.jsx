import renderer from 'react-test-renderer'
import { Circle } from '../../components/ui/circle/circle'
import { ElementStates } from '../../types/element-states'
import React from 'react'

describe('check circle', () => {
    const reactElement = <div>srtg</div>
    const testCases = [
        {
            letter: undefined,
            head: null,
            tail: null,
            index: undefined,
            isSmall: false,
            state: ElementStates.Default,
            explation: 'без буквы, head, tail, index и в состоянии Default'
        },
        {
            letter: 'strg',
            head: 'head',
            tail: 'tail',
            index: 0,
            isSmall: false,
            state: ElementStates.Changing,
            explation: 'с буквами, head, tail, index и в состоянии Changing'
        },
        {
            letter: 'strg',
            head: reactElement,
            tail: reactElement,
            index: 0,
            isSmall: false,
            state: ElementStates.Modified,
            explation: 'с буквами, с react-элементом в head, tail и в состоянии Modified'
        },
        {
            letter: 'strg',
            head: null,
            tail: null,
            index: undefined,
            isSmall: true,
            state: ElementStates.Default,
            explation: 'с буквами, с пропсом isSmall === true'
        }]
    testCases.forEach((attribute) => {
        it(`Кружок ${attribute.explation} рендерится без ошибок`, () => {
            const circle = renderer.create(<Circle letter={attribute.letter}
                head={attribute.head}
                tail={attribute.tail}
                index={attribute.index}
                isSmall={attribute.isSmall}
                state={attribute.state} />)
            expect(circle).toMatchSnapshot()
        })

    })

})