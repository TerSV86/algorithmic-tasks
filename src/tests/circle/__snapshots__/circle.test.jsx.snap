// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check circle Кружок без буквы, head, tail, index и в состоянии Default рендерится без ошибок 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="circle-state"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`check circle Кружок с буквами, head, tail, index и в состоянии Changing рендерится без ошибок 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   changing"
    data-testid="circle-state"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      strg
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 string"
  >
    tail
  </div>
</div>
`;

exports[`check circle Кружок с буквами, с react-элементом в head, tail и в состоянии Modified рендерится без ошибок 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      srtg
    </div>
  </div>
  <div
    className="circle   modified"
    data-testid="circle-state"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      strg
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  >
    <div>
      srtg
    </div>
  </div>
</div>
`;

exports[`check circle Кружок с буквами, с пропсом isSmall === true рендерится без ошибок 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="circle-state"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      strg
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
